addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -0,0 +1,825 @@
+		sendCommand(mac + " playlist delete " + playlistIndex);
+	}
+
+	public void playPlaylistItem(String mac, int playlistIndex) {
+		sendCommand(mac + " playlist index " + playlistIndex);
+	}
+
+	public void addPlaylistItem(String mac, String url) {
+		sendCommand(mac + " playlist add " + url);
+	}
+
+	public void setPlayingTime(String mac, int time) {
+		sendCommand(mac + " time " + time);
+	}
+
+	public void setRepeatMode(String mac, int repeatMode) {
+		sendCommand(mac + " playlist repeat " + repeatMode);
+	}
+
+	public void setShuffleMode(String mac, int shuffleMode) {
+		sendCommand(mac + " playlist shuffle " + shuffleMode);
+	}
+
+	public void volumeUp(String mac, int currentVolume) {
+		setVolume(mac, currentVolume + VOLUME_CHANGE_SIZE);
+	}
+
+	public void volumeDown(String mac, int currentVolume) {
+		setVolume(mac, currentVolume - VOLUME_CHANGE_SIZE);
+	}
+
+	public void setVolume(String mac, int volume) {
+		if (0 > volume) {
+			volume = 0;
+		} else if (volume > 100) {
+			volume = 100;
+		}
+		sendCommand(mac + " mixer volume " + String.valueOf(volume));
+	}
+
+	public void showString(String mac, String line) {
+		showString(mac, line, 5);
+	}
+
+	public void showString(String mac, String line, int duration) {
+		sendCommand(mac + " show line1:" + line + " duration:"
+				+ String.valueOf(duration));
+	}
+
+	public void showStringHuge(String mac, String line) {
+		showStringHuge(mac, line, 5);
+	}
+
+	public void showStringHuge(String mac, String line, int duration) {
+		sendCommand(mac + " show line1:" + line + " font:huge duration:"
+				+ String.valueOf(duration));
+	}
+
+	public void showStrings(String mac, String line1, String line2) {
+		showStrings(mac, line1, line2, 5);
+	}
+
+	public void showStrings(String mac, String line1, String line2, int duration) {
+		sendCommand(mac + " show line1:" + line1 + " line2:" + line2
+				+ " duration:" + String.valueOf(duration));
+	}
+
+	/**
+	 * Send a generic command to a given player
+	 * 
+	 * @param playerId
+	 * @param command
+	 */
+	public void playerCommand(String mac, String command) {
+		sendCommand(mac + " " + command);
+	}
+
+	/**
+	 * Ask for player list
+	 */
+	public void requestPlayers() {
+		sendCommand("players 0");
+	}
+
+	/**
+	 * Send a command to the Squeeze Server.
+	 */
+	private synchronized void sendCommand(String command) {
+		if (!isConnected()) {
+			logger.debug("No connection to SqueezeServer, will attempt to reconnect now...");
+			connect();
+			if (!isConnected()) {
+				logger.error(
+						"Failed to reconnect to SqueezeServer, unable to send command {}",
+						command);
+				return;
+			}
+		}
+		logger.debug("Sending command: {}", command);
+		try {
+			BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(
+					clientSocket.getOutputStream()));
+			writer.write(command + NEW_LINE);
+			writer.flush();
+		} catch (IOException e) {
+			logger.error("Error while sending command to Squeeze Server ("
```

Comment:
You should probably set the Thing to OFFLINE (with COMMUNICATION_ERROR) here?
And you can then remove the logging (as the thing status change will be logged by the framework anyhow)